cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicHttpServiceContext.java

```diff
@@ -253,4 +259,16 @@ public void releaseCallResources() {
 
     return evaluated;
   }
+
+  @Override
+  public Map<String, String> getPreferences(EntityId entityId, boolean resolved) {
+    try {
+      return preferencesFetcher.get(entityId, resolved).getProperties();
+    } catch (NotFoundException | IOException e) {
+      // We wrap this in a RuntimeException as the NotFoundException class is not available in the package for
+      // the superclass of this class (cdap-system-app-api)
+      LOG.error("Unable to get preferences", e);
+      throw new RuntimeException("Unable to get preferences", e);
```

Comment:
being forced to throw RuntimeException is normally a sign that the API could use improvement, as it makes it very difficult for the caller to handle errors properly.

Reply:
The problem here is that `NotFoundException` lives in the `cdap-common` package and `SystemHttpServiceContext` lives in the `cdap-system-app-api` package which is itself one of the dependencies of `cdap-common`. We have a chicken-and-egg problem here.

I agree that wrapping this exception is not ideal, but other than creating yet another wrapper exception class in the `cdap-system-app-api` package, I don't think there's a good solution to this.